lib/iris/tests/unit/fileformats/netcdf/test_save.py

```diff
@@ -59,6 +59,17 @@ def test_attributes_arrays(self):
             ds.close()
         self.assertArrayEqual(res, np.arange(2))
 
+    def test_no_special_attribute_clash(self):
+        c1 = Cube([0], var_name='test', attributes={'name': 'bar'})
+        c2 = Cube([0], var_name='test_1', attributes={'name': 'bar_1'})
```

Comment:
That sounds about right actually. Perhaps a comment to that effect would be of value. I guess you could also check the second cube's attribute too for completeness, I'll leave that up to you.